torch/quantization/fx/quantize.py

```diff
@@ -1050,19 +1050,28 @@ def record_match(pattern, node, matched):
             if node.name not in match_map and node.name not in all_matched:
                 for pattern, value in patterns.items():
                     if is_match(modules, node, pattern):
+                        skip_this_match = False
                         if value is BinaryOp:
                             use_copy_node = all_node_args_have_no_tensors(node)
                             if use_copy_node:
                                 value = CopyNode  # type: ignore
-                        matched: List[Any] = []
-                        record_match(pattern, node, matched)
-                        for n in matched:
-                            match_map[n.name] = (
-                                node, matched, pattern, value(self, node),  # type: ignore
-                                self.qconfig_map[n.name])
-                            all_matched.add(n.name)
-                        # break after finding the first match
-                        break
+
+                            this_node_qconfig = self.qconfig_map[node.name]
```

Comment:
here we are coupling matching with quantize handler I think

Reply:
Correct.  It looks like the current matching strategy of a match being determined only by the node type and nothing else does not scale to some of these new requirements (this PR, and the PR earlier in this stack).  We now need to match by node type, dtype, and the existence of tensors (versus primitives) in the node's args.

My understanding of the high level concepts:
* `QuantizeHandler`: determines how a node or a fusion pattern should go through prepare and convert steps
* matching: matches tuples of nodes to corresponding quantize handler instances

One way to clean this up in the future is to remove the restriction that a single pattern can only match to a single instance of quantize handler, and make the matching process aware of this and do the match correctly with the extra information.

Reply:
I think it makes sense to extend matching to consider qconfig as well, but currently we don't do that. sounds good to me if we plan to clean this up later.

Reply:
maybe add some comments to say that we'll extend matching to consider qconfigs and clean this up?